data/core/macros/abilities.cfg

```diff
@@ -277,6 +341,51 @@ Enemy units cannot see this unit while it is in deep water, except if they have
     [/dummy]
 #enddef
 
+#define ABILITY_FORMATION
+    [dummy]
+        id=formation
+        name= _ "formation"
+        female_name= _ "female^formation"
+        description= _ "This unit gains a +10% bonus to defense when another unit with the same ability is adjacent to it. However, this cannot raise the unit’s defense above 70%."
+        [filter]
+            [filter_adjacent]
+                ability=formation
+                is_enemy=no
+                count=1-5
+            [/filter_adjacent]
+        [/filter]
+    [/dummy]
+    {FORMATION formation1 40 1-5}
+    {FORMATION formation2 50 2-5}
+    {FORMATION formation3 60 3-5}
+    {FORMATION formation4 70 4-5}
+    {FORMATION formation5 80 5}
+#enddef
+
+#define ABILITY_DISENGAGE
+    [dummy]
+        id=disengage
+        name= _ "disengage"
+        female_name= _ "female^disengage"
+        description= _ "If this unit doesn’t move before attacking, it will retain its movement points after the attack."
+    [/dummy] # wmlxgettext: [abilities]
+[/abilities]
+{ABILITY_DISENGAGE_EVENTS}
+[+abilities] # wmlxgettext: [/abilities]
```

Comment:
Macros with unbalanced tags is a deprecated way to add ability-events.

\todo improve this comment to say what the right way to do it is, when I remember. If someone else reviews, please comment.

Reply:
WEAPON_SPECIAL_DAZE_EVENTS is a reason for me to be not favorable for move quenoth elves to mainline, because unit placed event put an [object] to many units, if you're use this implementation, it is active only if a unit with WEAPON_SPECIAL_DAZE is present on the map, but this weapon can't be added via [effect], and if you change for implement this  in specific campaign or era, then this units can be used in other secenario of mainline, but daze will be not efficient.